opentelemetry-sdk/src/metrics/data/mod.rs

```diff
@@ -38,23 +38,77 @@ pub struct Metric {
     /// The unit in which the instrument reports.
     pub unit: Cow<'static, str>,
     /// The aggregated data from an instrument.
-    pub data: Box<dyn Aggregation>,
+    pub data: AggregatedMetrics,
 }
 
-/// The store of data reported by an [Instrument].
-///
-/// It will be one of: [Gauge], [Sum], or [Histogram].
-///
-/// [Instrument]: crate::metrics::Instrument
-pub trait Aggregation: fmt::Debug + any::Any + Send + Sync {
-    /// Support downcasting
-    fn as_any(&self) -> &dyn any::Any;
-    /// Support downcasting during aggregation
-    fn as_mut(&mut self) -> &mut dyn any::Any;
+/// Aggregated metrics data from an instrument
+#[derive(Debug, Clone)]
+pub enum AggregatedMetrics {
+    /// All metric data with `f64` value type
+    F64(MetricData<f64>),
+    /// All metric data with `u64` value type
+    U64(MetricData<u64>),
```

Comment:
this shouldn't exist. Only f64 and i64 should ever come out after aggregation.

Reply:
.. but `InstrumentProvider` has `u64_counter`, `u64_gauge`, `u64_histogram`, etc...

Reply:
There are no conversions happening at the moment, e.g. in case a user creates `u64_counter`, we'll return `MetricData<u64>`. I'm not sure at which point we should convert from `u64` to `i64`.
Maybe let's leave this for another discussion?

Reply:
yes lets follow up separately.